### ISSUE TITLE
META files are not installed for coq libraries

### ISSUE THREAD

**Issue by zorun** [coq/coq#7296]
#### Version

8.7.2

#### Operating system

Arch Linux

#### Description of the problem

This is a forward of this [Arch Linux bug report](https://bugs.archlinux.org/task/58203) because it looks like an upstream issue.

> The libraries installed with package coq are not registered with ocamlfind, making them essentially  unusable for most software that build-depends on coq. It seems that registering libraries with ocamlfind is also recommended practice (https://wiki.archlinux.org/index.php/OCaml_package_guidelines#Libraries).
> Steps to reproduce:

```
$ pacman -Si coq | grep -i version :(
Version : 8.7.2-1
$ pacman -Ql coq | grep -q META || printf "not META\n"
not META
$ ocamlfind list | grep -i coq || printf "not found\n"
not found
$ git clone https://github.com/ejgallego/coq-serapi.git
$ cd coq-serapi
$ make 2>&1 > /dev/null
$ ocamlbuild -use-ocamlfind -j 4 sertop/sertop.native
+ ocamlfind ocamldep -package cmdliner -package coq.intf,coq.stm,sexplib -modules sertop/sertop.ml > sertop/sertop.ml.depends
ocamlfind: Package `coq.intf' not found
Command exited with code 2.
Compilation unsuccessful after building 1 target (0 cached) in 00:00:00.
```

**Comment by ejgallego** [comment]
Certainly Coq's makefile does install META in the proper place:
https://github.com/coq/coq/blob/v8.7/Makefile.install#L163

So this looks like a distro-specific problem.

**Comment by ejgallego** [comment]
Also note that the default branch for coq-serapi is 8.8, you wanna do:

> git clone -b v8.7 https://github.com/ejgallego/coq-serapi.git

in order to fix that. Meanwhile, you can add to your OCAMLPATH the location of `META.coq` that the Arch package does install.

**Comment by maximedenes** [comment]
Is there any reason not to call `make install`?

**Comment by zorun** [comment]
@maximedenes yes, the goal is to install coq and coqide in seperate packages, see: https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/coq#n47

**Comment by maximedenes** [comment]
Ah, thanks! @ejgallego what are your current thoughts about splitting off CoqIDE? I believe you've tried it in the past.

**Comment by ejgallego** [comment]
@maximedenes the PR doing the last steps to make this possible has been sitting on my hard disk for a while... no point on submitting until some other PRs are merged, but part of it is included in #6857 if you want a peek preview.

**Comment by maximedenes** [comment]
Ok, but you still think it is a desirable goal? And concretely, does it mean moving CoqIDE to a separate repository, or only making the builds independent?

**Comment by ejgallego** [comment]
In particular, I plan to do as follows: provide a `_tags` file that allows building CoqIDE from an installed form in the way of https://github.com/ejgallego/coqide-exp , and keep the makefile as-is, so people can configure without CoqIDE, then call the `_tags` target.

Also, most distributors would prefer to use the Dune build method when available, for that you only need to do `jbuild -p coqide`

**Comment by zorun** [comment]
For Arch packaging purpose, it's fine to have a common build steps and then separate install targets for coq and coqide.

I didn't know about Dune, but it's already used by some Arch packages so it should be fine to support.

**Comment by ejgallego** [comment]
Dune has two advantages:
- it manages the install lists automatically
- it transparently support in-tree / out-of-tree packages.
